File: Flutter-app/lib/models/benchmark_result.dart
class BenchmarkResult {
  final String scenario;
  final double executionTimeMs;
  final double cpuUsagePercent;
  final double memoryUsageMb;
  final double memoryMb;
  final double cpuPercent;
  final DateTime timestamp;

  BenchmarkResult({
    required this.scenario,
    required this.executionTimeMs,
    this.cpuUsagePercent = 0.0,
    this.memoryUsageMb = 0.0,
    this.memoryMb = 0.0,
    this.cpuPercent = 0.0,
    DateTime? timestamp,
  }) : timestamp = timestamp ?? DateTime.now();
}

File: Kotlin-app/app/src/main/java/com/benchmark/androidnative/model/BenchmarkResult.kt
package com.benchmark.androidnative.model

import java.util.Date

data class BenchmarkResult(
    val run: Int,
    val framework: String = "kotlin",
    val scenario: String,
    val executionTimeMs: Double,
    val cpuPercent: Double,
    val memoryMb: Double,
    val timestamp: Date = Date(),
)
